### PR TITLE
r/appsync_graphql_api: support additional_authentication_providers

### LINKED ISSUE #8715: Support additional authentication providers in Appsync
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Announced [here](https://aws.amazon.com/about-aws/whats-new/2019/05/aws-appsync-now-supports-configuring-multiple-authorization-type/), Appsync now supports specifying additional auth methods. The aws_appsync_graphql_api resource should be updated to allow configuring of these additional auth methods.

### New or Affected Resource(s)

* aws_appsync_graphql_api

### Potential Terraform Configuration

```hcl
resource "aws_appsync_graphql_api" "example" {
  authentication_type = "AWS_IAM"
  name = "example"

  additional_authentication_provider {
    authentication_type = "API_KEY"
  }
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2019/05/aws-appsync-now-supports-configuring-multiple-authorization-type/

There's a PR open already that should fix this issue: #8587 Opening an issue for greater visibility/tracking.

### PR DESCRIPTION
**Pull request by acburdine**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #8715 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
```release-note
resource/aws_appsync_graphql_api: support `additional_authentication_providers`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSAppsyncGraphqlApi'
--- PASS: TestAccAWSAppsyncGraphqlApi_basic (16.66s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AuthenticationType_AWSIAM (16.94s)
--- PASS: TestAccAWSAppsyncGraphqlApi_LogConfig (18.77s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AdditionalAuthentication_CognitoUserPools (19.31s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AdditionalAuthentication_Multiple (19.57s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AdditionalAuthentication_APIKey (23.70s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AdditionalAuthentication_AWSIAM (24.62s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AuthenticationType_AmazonCognitoUserPools (25.69s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AuthenticationType_APIKey (25.72s)
--- PASS: TestAccAWSAppsyncGraphqlApi_OpenIDConnectConfig_ClientID (26.03s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AdditionalAuthentication_OpenIDConnect (26.68s)
--- PASS: TestAccAWSAppsyncGraphqlApi_Name (26.96s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AuthenticationType_OpenIDConnect (27.96s)
--- PASS: TestAccAWSAppsyncGraphqlApi_UserPoolConfig_AwsRegion (29.07s)
--- PASS: TestAccAWSAppsyncGraphqlApi_disappears (12.64s)
--- PASS: TestAccAWSAppsyncGraphqlApi_OpenIDConnectConfig_IatTTL (35.69s)
--- PASS: TestAccAWSAppsyncGraphqlApi_OpenIDConnectConfig_Issuer (36.28s)
--- PASS: TestAccAWSAppsyncGraphqlApi_OpenIDConnectConfig_AuthTTL (36.64s)
--- PASS: TestAccAWSAppsyncGraphqlApi_UserPoolConfig_DefaultAction (37.32s)
--- PASS: TestAccAWSAppsyncGraphqlApi_Tags (37.93s)
--- PASS: TestAccAWSAppsyncGraphqlApi_LogConfig_FieldLogLevel (39.45s)
--- PASS: TestAccAWSAppsyncGraphqlApi_AuthenticationType (26.14s)
--- PASS: TestAccAWSAppsyncGraphqlApi_Schema (29.58s)
PASS
```

### CODE REVIEW

File: website/docs/r/appsync_graphql_api.html.markdown [hunk 1]
```diff
@@ -143,7 +165,7 @@ The following arguments are supported:
 
 The following arguments are supported:
 
-* `default_action` - (Required) The action that you want your GraphQL API to take when a request that uses Amazon Cognito User Pool authentication doesn't match the Amazon Cognito User Pool configuration. Valid: `ALLOW` and `DENY`
+* `default_action` - (Required only if Cognito is used as the default auth provider) The action that you want your GraphQL API to take when a request that uses Amazon Cognito User Pool authentication doesn't match the Amazon Cognito User Pool configuration. Valid: `ALLOW` and `DENY`
```
Comment (acburdine) [inline website/docs/r/appsync_graphql_api.html.markdown]: this particular bit probably needs better wording - the `default_action` parameter isn't supported in the additional_auth_provider's user_pool_config block, but I wasn't sure if this is the better way, or if I should create an entirely separate user_pool_config block specifically for the additional_authentication_provider.

**Comment by acburdine** [comment]
other than the minor documentation question - this should be good to go now

**Comment by acburdine** [comment]
@bflad apologies for the direct ping, wasn't sure how else to get this looked at 😕 is there any way I could help get this merged in?

**Comment by Vlaaaaaaad** [comment]
Is there anything blocking this?

**Comment by tgroshon** [comment]
Does this need any additional work?  This is very important to a project I am working on.

**Comment by rplute** [comment]
Why is this still open and not merged in yet?

**Comment by falcon47** [comment]
Hey guys, is there any plans to merge this soon? Any updates?

**Comment by aeschright** [comment]
Hi everyone, we're still working through the sizable backlog of PRs and have new team members ramping up. We'll get to this one as soon as we can, and you can help by making sure you've addressed the [common review items](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#common-review-items). Thanks!

**Comment by tgroshon** [comment]
Everyone is awesome here

**Pull request merged**